db/migrate/20160323133253_remove_unique_user_constraint.rb

```diff
@@ -0,0 +1,22 @@
+class RemoveUniqueUserConstraint < ActiveRecord::Migration
+  def up
+    remove_index :users, :name
+
+    # This column require the space for <human|technical>[-<organizationNormalizedName>]-<name.downcase>
+    add_column :users, :normalized_name, :string, null: true, limit: 100
+
+    add_index :users, :normalized_name, unique: true
+
+    User.all.each do |user|
+      user.save
```

Comment:
Avoid doing this in the future unless there's no other choice. Always prefer `update_all`.

It's dangerous because the implementation can change between the time the migration is written and the time it is run. It also updates all the users' timestamps, which I don't think we should do in cases like this.

We won't re-run it on trial, but you can rewrite it for demo.

```
User.all.each do |user|
  normalized_name = ... # hard-code implementation here
  User.where(id: user.id).update_all(normalized_name: normalized_name)
end
```